### PR TITLE
[docs] Update docs on where response extensions can be mutated

### PR DESCRIPTION
**Pull request by smyrick**
I couldn't find any articles or docs on how to do this as I was attempting myself to debug an issue for a customer

### CODE REVIEW

File: docs/source/integrations/plugins-event-reference.mdx [hunk 1]
```diff
@@ -487,6 +487,8 @@ if the GraphQL operation encounters one or more errors.
 
 (If the operation uses [incremental delivery](../workflow/requests#incremental-delivery-experimental) directives such as `@defer`, `willSendResponse` is called before the *initial* payload is sent; you can use `willSendSubsequentPayload` to find out when more payloads will be sent.)
 
+This is also the place you can mutate the GraphQL response in-place, like adding values to the `extensions` or merging responses from subgraphs with conditional logic.
```
Comment (glasser) [inline docs/source/integrations/plugins-event-reference.mdx:490]: I like having this addition in general. I don't know if talking about subgraphs in this Apollo Server context makes sense though (I'm not sure what the use case you're describing is).

Comment (smyrick) [reply docs/source/integrations/plugins-event-reference.mdx:490]: I suppose I can remove the subgraph reference here. That was more a callout to Gateway

File: docs/source/integrations/plugins-event-reference.mdx [hunk 2]
```diff
@@ -487,6 +487,8 @@ if the GraphQL operation encounters one or more errors.
 
 (If the operation uses [incremental delivery](../workflow/requests#incremental-delivery-experimental) directives such as `@defer`, `willSendResponse` is called before the *initial* payload is sent; you can use `willSendSubsequentPayload` to find out when more payloads will be sent.)
 
+This is also the place you can mutate the GraphQL response in-place, like adding values to the `extensions` or redacting information in the `error` blocks.
```
Comment (Meschreiber) [inline docs/source/integrations/plugins-event-reference.mdx:490]: It's better to not start new paragraphs with "this" to be clearer about what "this" means.  Is this accurate:?

```suggestion
The `willSendResponse` event is also where you can mutate the GraphQL response in-place. For example, you can add values to the `extensions` object or redact information in the `error` blocks.
```

**Review by Meschreiber — APPROVED** [review]

File: docs/source/integrations/plugins-event-reference.mdx [hunk 3]
```diff
@@ -487,6 +487,8 @@ if the GraphQL operation encounters one or more errors.
 
 (If the operation uses [incremental delivery](../workflow/requests#incremental-delivery-experimental) directives such as `@defer`, `willSendResponse` is called before the *initial* payload is sent; you can use `willSendSubsequentPayload` to find out when more payloads will be sent.)
 
+The `willSendResponse` event is also where you can mutate the GraphQL response in-place. For example, you can add values to the `extensions` object or redact information in the `error` blocks.
```
Comment (trevor-scheer) [inline docs/source/integrations/plugins-event-reference.mdx:490]: Shouldn't the latter be handled by [formatError](https://www.apollographql.com/docs/apollo-server/data/errors/#for-client-responses)?

**Review by trevor-scheer — CHANGES_REQUESTED** [review]

File: docs/source/integrations/plugins-event-reference.mdx (hunk 3 above)
Comment (smyrick) [reply docs/source/integrations/plugins-event-reference.mdx:490]: Good callout. I will update to reference that section

**Review by trevor-scheer — APPROVED** [review]

**Pull request merged**